### DISCUSSION TITLE
Highly correlated attributes with frequency adjustment (Address)

### DISCUSSION THREAD

**Discussion by niquola** [moj-analytical-services/splink#2348] [General]
In our case records have correlated attributes related to address:
* postal code
* line1/line2 
* city
* home phone

Records may have some of the attributes missed. 
What could be the good comparison levels (ideally with frequencies) 
or any recommendations on how to choose between attributes?

How can I measure the attributes correlation?

**Comment by RobinL** [comment]
Measuring is difficult, but the correlation in the address will be strong.  Correlation to home phone number shouldn't be so bad, so I would suggest two comparisons, one for address and the other for phone

I would suggest something like:

Address comparison, made up of levels
- Exact match on postal code and line 1 (possibly also include line 2).  If you want tf adjustments, you'll need to concatenate these into a 'full address' field prior to importing the data and match on that
- Levenshtein on the same full address field, with edit distance of no more than (say) 2
- Exact match on postal code, with tf adjustments on postcal code
- Levenshtein on postal code with edit distance of no more than 1
- Else (all others)

Then probably a standard [LevenshteinAtThresholds](https://moj-analytical-services.github.io/splink/api_docs/comparison_library.html#splink.comparison_library.LevenshteinAtThresholds) should do for the phone number

This is a fairly simple approach; more generally address matching is notoriously hard.  There's a more sophisticated example in the following repo of address matching (where address is the entity type).  

https://github.com/RobinL/uk_address_matcher

Worth noting that in most of our pipelines, because of the complexity of address matching, we tend to just match on postcode

**Reply by niquola** [reply]
Thank you! Can you elaborate on why "it is ok" to have highly correlated home phone and address as attributes?

**Reply by RobinL** [reply]
Do you think they're highly correlated? That would be the case if knowing the address means you know the phone number and/or vice versa.  I guess that's possibly the case with landlines, but wouldn't be the case with mobiles.  I guess I'd they're all landlines you could have an exact match on full address, and then a second comparison level for an exact match on phone number, all within the same comparison

**Reply by niquola** [reply]
It looks like it is a landline phones. I tested  by picking all  address/phone pairs  (on 4 years 4M dataset) and counting how often the pair is the same vs not - getting 50-70% for same - i.e I can with 50% probability predict address by phone or vice versa
